static/app/views/performance/browser/resources/utils/useResourcesQuery.ts

```diff
@@ -115,3 +112,28 @@ export const useResourcesQuery = ({sort, defaultResourceTypes, query, limit}: Pr
 
   return {...result, data: data || []};
 };
+
+export const getDomainFilter = (selectedDomain: string | undefined) => {
+  let domainFilter: string[] = [];
+  if (selectedDomain) {
+    if (selectedDomain === EMPTY_OPTION_VALUE) {
+      domainFilter = [`!has:${SPAN_DOMAIN}`];
+    } else {
+      domainFilter = [`${SPAN_DOMAIN}:${selectedDomain}`];
+    }
+  }
+  return domainFilter;
```

Comment:
let's avoid mutation + nesting if's. Ditto with `getResourceTypeFilter`

```suggestion
  if (!selectedDomain) {
    return [];
  }
 
  if (selectedDomain === EMPTY_OPTION_VALUE) {
    return [`!has:${SPAN_DOMAIN}`];
  }
  
  return [`${SPAN_DOMAIN}:${selectedDomain}`];
```

Reply:
👍